PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/appservice/AppServiceCreationComposite.java

```diff
@@ -96,29 +88,38 @@ protected Control createDialogArea(Composite parent) {
 
         });
         subscriptionComboBox.refreshItems();
-        return container;
+    }
+
+    public void setRuntime(final List<Runtime> platformList) {
+        instanceDetailPanel.getRuntimeComboBox().setPlatformList(platformList);
+    }
+
+    @Override
+    protected void checkSubclass() {
     }
 
     @Override
-    public AppServiceConfig getValue() {
+    public T getValue() {
         final AppServicePlanEntity entity = appServicePlanPanel.getServicePlan();
-        return new AppServiceConfig().subscriptionId(subsAndResourceGroupPanel.getSubscription().getId())
-                .resourceGroup(subsAndResourceGroupPanel.getResourceGroup().getName())
-                .appName(instanceDetailPanel.getAppName())
-                .region(instanceDetailPanel.getResourceRegion())
-                .runtime(instanceDetailPanel.getRuntime())
-                .servicePlanName(entity.getName())
-                .servicePlanResourceGroup(StringUtils.firstNonBlank(entity.getResourceGroup(),
-                        subsAndResourceGroupPanel.getResourceGroup().getName()))
-                .pricingTier(entity.getPricingTier());
+        final T result = supplier.get();
+        result.subscriptionId(subsAndResourceGroupPanel.getSubscription().getId());
```

Comment:
Have you considered the case subsAndResourceGroupPanel.getSubscription() returns null